packages/database/src/core/sync/buildSyncLookupTraverseJoins.js

```diff
@@ -0,0 +1,22 @@
+export function buildSyncLookupTraverseJoins(tablesToTraverse, customForeignKeys) {
```

Comment:
These build sync_lookup utils originally live in `@tupaia/sync` package, because I was trying to encapsulate as much sync logic to @tupaia/sync package as possible. However, it causes a circular dependency problem because:
- @tupaia/sync imports @tupaia/database to use some Database types - ModelRegistry, TupaiaDatabase,...
- @tupaia/database uses @tupaia/sync for some of the sync utils

I decided to move some database related sync utils to live in @tupaia/database package itself to resolve the circular dependency issue. I also think it makese sense for the @tupaia/database package to be aware of some of the sync logic